src/validation/coinbase.cairo

```diff
@@ -31,7 +33,17 @@ pub fn validate_coinbase(
     assert(total_output_amount <= total_fees + block_reward, 'total output > block rwd + fees');
 
     // TODO: validate BIP-34 sig script
-    // TODO: validate BIP-141 witness field
+
+    // validate BIP-141 witness field
+    if block_height >= BIP_141_BLOCK_HEIGHT {
+        let witness_reserved_value: ByteArray =
+            "0000000000000000000000000000000000000000000000000000000000000000";
+        let witness = *tx.inputs[0].witness;
+
```

Comment:
Return error instead of assert
Also check that witness array contains only one element